### PR TITLE
Fix failing AsyncFunction tests in test262-esnext

### PR DESCRIPTION
**Pull request by pmarkee**
JerryScript-DCO-1.0-Signed-off-by: Peter Marki marpeter@inf.u-szeged.hu

### CODE REVIEW

**Review by rerobika — APPROVED** [review]

**Review by galpeter — APPROVED** [review]
lgtm

File: jerry-core/parser/js/byte-code.h [hunk 1]
```diff
@@ -907,10 +907,10 @@ typedef enum
   /* The following functions cannot be constructed (see CBC_FUNCTION_IS_CONSTRUCTABLE) */
   CBC_FUNCTION_SCRIPT, /**< script (global) function */
   CBC_FUNCTION_GENERATOR, /**< generator function */
-  CBC_FUNCTION_ASYNC, /**< async function */
   CBC_FUNCTION_ASYNC_GENERATOR, /**< async generator function */
 
   /* The following functions has no prototype (see CBC_FUNCTION_HAS_PROTOTYPE) */
+  CBC_FUNCTION_ASYNC, /**< async function */
```
Comment (zherczeg) [inline jerry-core/parser/js/byte-code.h:913]: I would put this below CBC_FUNCTION_ACCESSOR

**Review by zherczeg — CHANGES_REQUESTED** [review]

File: jerry-core/parser/js/byte-code.h (hunk 1 above)
Comment (pmarkee) [reply jerry-core/parser/js/byte-code.h]: Done.

**Review by zherczeg — APPROVED** [review]
LGTM

**Pull request merged**